numba/np/random/generator_methods.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from numba.core import types
+from numba.core.extending import overload_method
+from numba.core.imputils import Registry
+from numba.np.numpy_support import as_dtype, from_dtype
+from numba.np.random.generator_core import next_float, next_double
+
+
+registry = Registry('generator_methods')
+
+
+def get_proper_func(func_32, func_64, dtype):
```

Comment:
This function will need some comments, to explain what is going on. Also, it may be wise to unit-test this, the function does seem to be quite straightforward to test.